lang_python/parsing/parser_python.mly

```diff
@@ -496,8 +496,16 @@ excepthandler:
   | EXCEPT test COMMA test COLON suite { ExceptHandler (Some $2, Some (expr_store $4), $6) }
 
 with_stmt:
-  | WITH test         COLON suite { With ($2, None, $4) }
-  | WITH test AS expr COLON suite { With ($2, Some $4, $6) }
+  | WITH with_list COLON suite { With ($2, $4) }
```

Comment:
Here, you could just process the with_list list and generate 
With (elem1 (With elem2 (With elem3 ...) until the end of the list which you would have $4
Basically something like   fold_right (fun acc e -> With e acc) $2 $4